src/ci/build.sh

```diff
@@ -18,33 +18,47 @@
 set -e
 
 COMMON_SCRIPT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/common.sh"
+# shellcheck source=src/ci/common.sh
 source "${COMMON_SCRIPT}"
 
-# all the prep is done, lets run the build!
-MVN_CMD="mvn -s src/ci/settings.xml"
+cron () {
+    echo "Running TRAVIS CRON task"
+    ${MVN_CMD} dependency-check:aggregate -Powasp
+}
+
+deploy () {
+    echo "Deploying SNAPSHOT build"
+    ${MVN_CMD} deploy -Pci
+
+    # also deploy the javadocs to the site
+    git clone -b gh-pages "https://github.com/${REPO_SLUG}.git" target/gh-pages/
+    ${MVN_CMD} javadoc:aggregate com.okta:okta-doclist-maven-plugin:generate jxr:aggregate -Ppub-docs -Pci
+}
+
+full_build () {
+    echo "Running mvn install"
+    ${MVN_CMD} install -Pci
+}
+
+no_its_build () {
+    echo "Skipping ITs, likely this build is a pull request from a fork"
+    ${MVN_CMD} install -DskipITs -Pci
+}
 
 # if this build was triggered via a cron job, just scan the dependencies
-if [ "$TRAVIS_EVENT_TYPE" = "cron" ] ; then
-    echo "Running TRAVIS CRON task"
-    $MVN_CMD dependency-check:aggregate -Powasp
+if [ "${TRAVIS_EVENT_TYPE}" = "cron" ] ; then
+    cron
 else
     # run 'mvn deploy' if we can
-    if [ "$DEPLOY" = true ] ; then
-        echo "Deploying SNAPSHOT build"
-        $MVN_CMD deploy -Pci
-
-        # also deploy the javadocs to the site
-        git clone -b gh-pages https://github.com/${REPO_SLUG}.git target/gh-pages/
-        $MVN_CMD javadoc:aggregate -Ppub-docs -Pci
+    if [ "${DEPLOY}" = true ] ; then
+        deploy
     else
         # else try to run the ITs if possible (for someone who has push access to the repo
-        if [ "$RUN_ITS" = true ] ; then
-            echo "Running mvn install"
-            $MVN_CMD install -Pci
+        if [ "${RUN_ITS}" = true ] ; then
+            full_build
```

Comment:
nit: Instead of breaking this into nested `if else` commands, I'd suggest evaluating the existance of this var (`${RUN_ITS}`) inside of the `full_build` command.

```bash
full_build () {
    echo "Running mvn install"
    if [ "${RUN_ITS}" = true ] ; then
      ${MVN_CMD} install -Pci
    else
      echo "Skipping ITs, likely this build is a pull request from a fork"
      ${MVN_CMD} install -DskipITs -Pci
}
```